docs/networkfence.md

```diff
@@ -0,0 +1,47 @@
+# NetworkFence
+
+NetworkFence is a cluster-scoped custom resource that allows Kubernetes to invoke "Network fence" operation on a storage provider.
+
+The user needs to specify the kind of operation they want to perform (fence/unfence), and then provide the list of CIDR blocks on which the mentioned operation is to be performed.
+This results in the fence/unfence of the target clients via NetworkFence custom resource.
+
+### Fence Operation
+
+```yaml
+apiVersion: csiaddons.openshift.io/v1alpha1
+kind: NetworkFence
+metadata:
+  name: network-fence-sample
+spec:
+  operation: fence
+  cidr:
```

Comment:
don't we need secrets to talk to the CSI driver? how it will be handled?

Reply:
Indeed, would it use the provisioner or node-plugin Secrets linked from a StorageClass, or maybe from a PV?

I guess it is possible to link to some secret, but that makes it not very user friendly.

This should be a privileged operation too. What is the plan to not allow anyone with permissions to the Kubernetes cluster to fence random CIDRs?

Reply:
> This should be a privileged operation too. What is the plan to not allow anyone with permissions to the Kubernetes cluster to fence random CIDRs?


Added secrets as a part of the CR now. Since blocking access is an admin operation, the `NetworkFence` CR creation will be done by high access users(admin most probably) and not by a common user.  The secrets used, thus, will have privileged access.